### DISCUSSION TITLE
Fooocus an alternative to abandoned Forge?

### DISCUSSION THREAD

**Discussion by Marco76DE** [lllyasviel/Fooocus#3244] [Q&A] [answered]
Hello dear people,as a former Automatic1111 Forge user I am looking for a new good web-ui.
I am not very familiar with Foocus,and I am not sure about its general abilities,or if its does support important features like vram-optimization,controlnet etc. I want to give it a try but i just found out that Fooocus is not supporting any extensions like reactor or other stuff.Is this the fact? If yes there is no reason at all for me to use Fooocus. Thx.

**Comment by bsdesigner** [comment, marked as answer]
Himself asked. Himself answered. Himself thanked. He himself went back to where he came from... )))

**Reply by Marco76DE** [reply]
Himself had not much time when creating this posting so himself decided to throw it out in a very "fast,raw,uncomplicated" way.
Himself is thanking you for your even more "compressed" answer. :-)
Himself went back to where he came from,to the wonderful place on earth where stable diffusion was developed by wonderful pioneers in germany.

**Reply by CantainPike** [reply]
Try SwarmUI, my first choice is Fooocus, I am also playing  SwarmUI there is a lot to discover in SwarmUI join their discord develeper very active and responsive

**Reply by mashb1t** [reply]
The biggest difference is modularity. Fooocus has very specialized workflows and isn't easily extensible, whereas other tools do offer a plugin system, which can't by design be as optimized.
This fact also leads to a high learning curve as developers new to Fooocus have to understand the structure as a whole rather than only parts of the code, which isn't convenient for starters.

**Reply by Marco76DE** [reply]
> The biggest difference is modularity. Fooocus has very specialized workflows and isn't easily extensible, whereas other tools do offer a plugin system, which can't by design be as optimized. This fact also leads to a high learning curve as developers new to Fooocus have to understand the structure as a whole rather than only parts of the code, which isn't convenient for starters.

Thx very much for this interesting explenation. It´s not easy to recognize such important things if you are not very familiar with the technical background of any system and what this means for developers etc.
